File: ext/src/main/Markup/Html/SgmlTag.c
/***************************************************************************
 *   Copyright (C) 2007 by Konstantin V. Arkhipov                          *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU Lesser General Public License as        *
 *   published by the Free Software Foundation; either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 ***************************************************************************/
/* $Id$ */

#include "onphp.h"

ONPHP_GETTER(SgmlTag, getId, id);
ONPHP_SETTER(SgmlTag, setId, id);

static ONPHP_ARGINFO_ONE;

zend_function_entry onphp_funcs_SgmlTag[] = {
	ONPHP_ME(SgmlTag, getId, NULL, ZEND_ACC_PUBLIC)
	ONPHP_ME(SgmlTag, setId, arginfo_one, ZEND_ACC_PUBLIC)
	{NULL, NULL, NULL}
};

File: src/Main/Markup/Html/SgmlTag.php
<?php
/***************************************************************************
 *   Copyright (C) 2007 by Ivan Y. Khvostishkov                            *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU Lesser General Public License as        *
 *   published by the Free Software Foundation; either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 ***************************************************************************/

namespace OnPHP\Main\Markup\Html;

/**
 * @ingroup Html
 * @ingroup Module
**/
abstract class SgmlTag extends SgmlToken
{
	/**
	 * @var string|null
	 */
	private ?string $id = null;

	/**
	 * @param string|null $id
	 * @return static
	 */
	public function setId(?string $id): SgmlTag
	{
		$this->id = $id;

		return $this;
	}

	/**
	 * @return string|null
	 */
	public function getId(): ?string
	{
		return $this->id;
	}
}